### ISSUE TITLE
Questions on LGBM's search space

### ISSUE THREAD

**Issue by stepthom** [microsoft/FLAML#171]
LGBM's search space includes `subsample`, [also known as](https://lightgbm.readthedocs.io/en/latest/Parameters.html#bagging_fraction) `bagging_fraction`:

https://github.com/microsoft/FLAML/blob/10082b9262a862ef51cf09117bc5d13e43d1ca0d/flaml/model.py#L224-L227

However, according to [LGBM documentation](https://lightgbm.readthedocs.io/en/latest/Parameters.html#bagging_fraction), `bagging_freq` (alias `subsample_freq`) must be set to a non-zero value in order for `subsample` to take effect. 

However, I do not believe `bagging_freq` is set in FLAML, so it will take on its default value of 0, meaning that searching over `subsample` will have no effect.

Also, LGBM's search space includes `log_max_bin`:

https://github.com/microsoft/FLAML/blob/10082b9262a862ef51cf09117bc5d13e43d1ca0d/flaml/model.py#L228-L231

but I could not find mention of that parameter in the documentation. [I did see](https://github.com/microsoft/FLAML/blob/10082b9262a862ef51cf09117bc5d13e43d1ca0d/flaml/model.py#L224-L227) a `max_bin`, but no `log_max_bin`.

**Comment by sonichi** [comment]
@stepthom Good catch about `bagging_freq`, we'll set it to 1 and test the performance in next version.

`log_max_bin` is just the logarithm of `max_bin`. It's converted to `max_bin` when passing to LGBM.

**Comment by stepthom** [comment]
Got it - thank you for confirming!
